### PR TITLE
feat: rewrite calls to `float()` as `_vectorizedFloat()`

### PR DESCRIPTION
**Pull request by onelson**
Fixes: #4735, #4603 

With this diff, the vectorization pass of the semantic graph will rewrite calls to `float()` such that they call `_vectorizedFloat()` instead (provided by #5015).

In order for the vectorization pass to type check in tests, the analyzer had to be configured to see the stdlib such that `float` would not give an "undefined identifier" error.
This was the same error which prevented boolean literals from vectorizing (in tests only, or so it seems) so #4997 is likely unblocked with these changes.

The function identifier rewrite is guarded by a feature flag: `vectorizedFloat`

### Checklist

Dear Author :wave:, the following checks should be completed (or explicitly dismissed) before merging.

- [x] ✏️ Write a PR description, regardless of triviality, to include the _value_ of this PR
- [x] 🔗 Reference related issues
- [x] 🏃 Test cases are included to exercise the new code
- [x] 🧪 If **new packages** are being introduced to stdlib, link to Working Group discussion notes and ensure it lands under `experimental/` **N/A**
- [x] 📖 If **language features** are changing, ensure `docs/Spec.md` has been updated **N/A**

Dear Reviewer(s) :wave:, you are responsible (among others) for ensuring the completeness and quality of the above before approval.

### CODE REVIEW

File: stdlib/universe/map_vectorize_float_test.flux
Comment (wolffcm) [inline stdlib/universe/map_vectorize_float_test.flux:71]: Does it make sense to add a call to the stdlib's `toFloat` function in here?

It would be nice if we moved the vectorization logic into a planner rule so we could assert on the rule enabling vectorization.

File: libflux/flux-core/src/semantic/tests/vectorize.rs
Comment (wolffcm) [inline libflux/flux-core/src/semantic/tests/vectorize.rs:30]: Is this inferring the whole stdlib every time we receive a request?

**Review by wolffcm — COMMENTED** [review]
Looks good to me. I just had a question about if we're inferring the stdlib when maybe we don't need to?

File: libflux/flux-core/src/semantic/tests/vectorize.rs
Comment (onelson) [reply libflux/flux-core/src/semantic/tests/vectorize.rs:30]: I think "yes" but more specifically, note that this is happening for regular scripts regardless.

This particular analyzer is used for these tests specific tests - the change here has no impact on the regular execution of flux otherwise.

Previously there was no need for the prelude (or any other part of the stdlib) to be visible during these test runs. Probably this was skipped 
- to be faster
- it was needless (before now)
- (my observation working on this issue) it creates a lot of noise if you're watching in a debugger

File: stdlib/universe/map_vectorize_float_test.flux
Comment (onelson) [reply stdlib/universe/map_vectorize_float_test.flux:71]: I was just looking at queries in lovelace to see what would be needed and I was _literally just thinking about this_. For lovelace I'm planning to have 2 queries, one with a hand-written cast (like you see here) and a 2nd relying on `toFloat()` instead.

I will expect to see the same perf increase for both when we flip the feature flag.

I'm happy to add a test here using `toFloat()` and verify we hit the vectorized path using the debugger before this merges.

File: libflux/flux-core/src/semantic/tests/vectorize.rs
Comment (onelson) [reply libflux/flux-core/src/semantic/tests/vectorize.rs:30]: Oh, and to speak to why this is needed -- without this (or something similar) vectorization will fail in these tests because the program will error immediately with `undefined identifier: float`.

Comment (wolffcm) [reply libflux/flux-core/src/semantic/tests/vectorize.rs:30]: Ah, ok. I missed that this was test code.

**Review by wolffcm — APPROVED** [review]
Looks great.

File: stdlib/universe/map_vectorize_float_test.flux
Comment (onelson) [reply stdlib/universe/map_vectorize_float_test.flux:71]: So, I wrote a test to compare a plain map against the tables produced by `toFloat()` and it turns out, the `toFloat()` call is _not_ vectorizing while the `map()` _is_. This is unexpected.

I can investigate further, but I wonder if this should hold up the PR from merging. I'd like to get a better understanding of why these act different just to strengthen my mental model.

File: libflux/flux-core/src/semantic/tests/vectorize.rs
Comment (onelson) [inline libflux/flux-core/src/semantic/tests/vectorize.rs:30]: @Marwes This is the change I made allowing my tests to include calls to `float`. If I understood you during standup, your recommendation instead is to effectively hand-write the builtin definition for the bits each test needs.

I'm not sure how that should be done. Any advice?

Comment (onelson) [reply libflux/flux-core/src/semantic/tests/vectorize.rs:30]: Huh, okay. Rewriting the flux source like the following, without the prelude configured does seem to work okay.

```
builtin float : (v: A) => float

(r) => ({ r with a: float(v: r._value) })
```

I'll make a change for this and make a note on #4997

Comment (Marwes) [reply libflux/flux-core/src/semantic/tests/vectorize.rs]: Yeah, there is also `imp` and `env` to let us set things like this up https://github.com/influxdata/flux/issues/4997#issuecomment-1216424229

Comment (onelson) [reply libflux/flux-core/src/semantic/tests/vectorize.rs]: Moved this up to where we initialize the analyzer in 889816d1d4d952864b6eb7034a3f10761ee9f660 -- feels more official to do it this way :+1:

File: stdlib/universe/map_vectorize_float_test.flux
Comment (onelson) [reply stdlib/universe/map_vectorize_float_test.flux:71]: I've opened https://github.com/influxdata/flux/issues/5102 to investigate this further, but it looks like we might just not be vectorizing anything that isn't defined in the main package right now.

Redefining `toFloat` in `main` like so, then using it, showed it hits the vectorized code path.

```
toFloat2 = (tables=<-) => tables |> map(fn: (r) => ({r with _value: float(v:r._value)}))
```

I'm guessing there are could be other functions in stdlib that could be vectorized but aren't.

**Pull request merged**
